### PR TITLE
Corrected URL of IPR Praha low-vegetation ortophoto

### PR DESCRIPTION
**Pull request by Vlada-C**
Correcting URL of the service, exact link to license and updating category

### CODE REVIEW

**Comment by cicku** [comment]
It looks like the domain changed, can you clarify the relations between them?

**Comment by grischard** [comment]
@Vlada-C could you please add a privacy policy url? The test currently fails:

> ERROR:root:sources/europe/cz/PrahaIPRlow-vegetationorthophoto.geojson has no privacy_policy_url. Adding privacy policies to sources is important to comply with legal requirements in certain countries.

**Comment by Vlada-C** [comment]
> It looks like the domain changed, can you clarify the relations between them?

Yes, the domain changed - see https://lists.openstreetmap.org/pipermail/talk-cz/2020-December/023063.html

Auto-translated to English: https://lists-openstreetmap-org.translate.goog/pipermail/talk-cz/2020-December/023063.html?_x_tr_sl=cs&_x_tr_tl=en&_x_tr_hl=cs&_x_tr_pto=wapp

mepnet.cz was a technical infrastructure ("metropolitan network") owned by the city of Prague. Mepnet has been phased out.  So the map was moved to the domain of Prague city itself. The actual "operator", "provider" or "maintainer" of the map is IPR Praha, which is a metropolitan planning institute under full control of Prague municipal office. This has not changed - IPR was responsible for the map already before the change of the domain.

**Comment by Vlada-C** [comment]
> @Vlada-C could you please add a privacy policy url? The test currently fails:
> 
> > ERROR:root:sources/europe/cz/PrahaIPRlow-vegetationorthophoto.geojson has no privacy_policy_url. Adding privacy policies to sources is important to comply with legal requirements in certain countries.

I have added the best privacy policy URL I could find...

**Comment by matheusgomesms** [comment]
Privacy policy looks good to me, and also it is time to fix these IPR Praha layers, which are of very high-quality and useful.

@Vlada-C, right now, looks good to me to be merged. But let's use this opportunity to improve all IPR Praha layers. My questions, as a ocasional mapper in Prague, are:

1 - what are the differences between the IPR layers and tmsproxy? Are these the same layers, but being a backup service in case IPR layers are down, maintained by OSM CZ?
2 - what are the differences between latest and low-vegetation layers?
3 - can we set one of these layers as "best", so they are the default layers for any mapper in iD? Or is Bing much better?

Thanks a lot for your help!

Edit: with the answer of question 2, we can improve description of these layers as well.

**Comment by Vlada-C** [comment]
Hi Matheus,

> ... But let's use this opportunity to improve all IPR Praha layers...

Yes, that's exactly what I would like to do. As I am new here, I wanted to fix one file only at first, to see the procedure and to make sure that I do not break anything. To be honest, after doing the same thing in JOSM, I expected a quicker progress here. Nevertheless, I am absolutely willing to comply with the rules and habits here, explaining and improving my input. If you prefer that we do all the fixes and improvements of the Prague imagery in one shot, I can try to amend the pull request further. I would like to unify the names of the TMS mirror layers with the original IPR layers too.

Actually, there is more than just the IPR fix to come. I intend to add another couple of WMS layers - including an excellent ortophoto of the whole Czech Republic - that has been newly available as open data from the CUZK. The OSM CZ have managed to get an explicit approval on top of a CC-BY license for those layers and we have already added them to the JOSM collection. I hope this can be done separately and I would prefer not to mix this discussion about IPR layers with that about the CUZK ones.

> 1 - what are the differences between the IPR layers and tmsproxy? Are these the same layers, but being a backup service in case IPR layers are down, maintained by OSM CZ?

Yes, the tmsproxy layers are the same two IPR layers cached on a server of an OSM CZ member.

> 2 - what are the differences between latest and low-vegetation layers?

Both are similar in terms of excellent resolution and excellent position accuracy. Each of them is updated in a yearly cycle. The low-vegetation imagery is taken after the leaves are fallen from the trees and it may actually be newer than the "latest" imagery during a certain period of the year.

> 3 - can we set one of these layers as "best", so they are the default layers for any mapper in iD? Or is Bing much better?

I would suggest that the "IPR low-vegetation" layer is the new "best" layer for Prague.

File: sources/europe/cz/PrahaIPRlow-vegetationorthophoto.geojson [hunk 1]
```diff
@@ -4,17 +4,18 @@
...
+        "url": "https://gs-pub.praha.eu/imgs/services/ort/mimovegetacni_letecke_snimkovani/ImageServer/WMSServer?FORMAT=image/jpeg&VERSION=1.1.1&SERVICE=WMS&REQUEST=GetMap&LAYERS=0&STYLES=&SRS={proj}&WIDTH={width}&HEIGHT={height}&BBOX={bbox}",
         "country_code": "CZ",
-        "license_url": "https://wiki.openstreetmap.org/wiki/Cs:WikiProjekt_%C4%8Cesko/freemap#Ortofotomapy_IPR_Praha",
+        "license_url": "https://wiki.openstreetmap.org/wiki/Cs:%C4%8Cesko/freemap#Mapov%C3%A9_podklady_IPR_Praha",
+        "privacy_policy_url": "https://iprpraha.cz/stranka/95/informace-o-zpracovani-osobnich-udaju",
         "available_projections": [
             "CRS:84",
             "EPSG:4326",
             "EPSG:5514"
         ],
```
Comment (matheusgomesms) [inline sources/europe/cz/PrahaIPRlow-vegetationorthophoto.geojson:18]:
```suggestion
        ],
        "attribution": {
            "url": "https://iprpraha.cz/",
            "text": "IPR Praha"
        },
        "best": true,
        "valid-georeference": true,
```

File: sources/europe/cz/PrahaIPRlow-vegetationorthophoto.geojson [hunk 2]
```diff
@@ -4,17 +4,18 @@
         "id": "PrahaIPRlow-vegetationorthophoto",
```
Comment (matheusgomesms) [inline sources/europe/cz/PrahaIPRlow-vegetationorthophoto.geojson:5]:
```suggestion
        "icon": "https://osmlab.github.io/editor-layer-index/sources/europe/cz/IPR_Praha.png",
        "id": "PrahaIPRlow-vegetationorthophoto",
```

File: sources/europe/cz/PrahaIPRlow-vegetationorthophoto.geojson [hunk 3]
```diff
@@ -4,17 +4,18 @@
...
         "name": "Praha IPR low-vegetation orthophoto",
```
Comment (matheusgomesms) [inline sources/europe/cz/PrahaIPRlow-vegetationorthophoto.geojson:6]:
```suggestion
        "name": "Praha IPR low-vegetation orthophoto",
        "description": "Yearly IPR Praha orthophotos, taken after the leaves are fallen from the trees",
```

File: sources/europe/cz/PrahaIPRlow-vegetationorthophoto.geojson [hunk 4]
```diff
@@ -4,17 +4,18 @@
...
         "type": "wms",
-        "url": "https://giswas1.mepnet.cz/arcgis/services/MAP/mimovegetacni_snimkovani_cache/ImageServer/WMSServer?LAYERS=0&STYLES=default&FORMAT=image/jpeg&CRS={proj}&WIDTH={width}&HEIGHT={height}&BBOX={bbox}&VERSION=1.3.0&SERVICE=WMS&REQUEST=GetMap",
+        "url": "https://gs-pub.praha.eu/imgs/services/ort/mimovegetacni_letecke_snimkovani/ImageServer/WMSServer?FORMAT=image/jpeg&VERSION=1.1.1&SERVICE=WMS&REQUEST=GetMap&LAYERS=0&STYLES=&SRS={proj}&WIDTH={width}&HEIGHT={height}&BBOX={bbox}",
         "country_code": "CZ",
-        "license_url": "https://wiki.openstreetmap.org/wiki/Cs:WikiProjekt_%C4%8Cesko/freemap#Ortofotomapy_IPR_Praha",
+        "license_url": "https://wiki.openstreetmap.org/wiki/Cs:%C4%8Cesko/freemap#Mapov%C3%A9_podklady_IPR_Praha",
```
Comment (matheusgomesms) [inline sources/europe/cz/PrahaIPRlow-vegetationorthophoto.geojson:12]:
```suggestion
        "permission_osm": "explicit",
        "license_url": "https://wiki.openstreetmap.org/wiki/Cs:%C4%8Cesko/freemap#Mapov%C3%A9_podklady_IPR_Praha",
```

**Comment by matheusgomesms** [comment]
Thank you for your details, @Vlada-C!

Indeed, this should be much easier and faster, and I appreciate your willingness to help and improve ELI.

I've taken the liberty of improving your submission, so feel free to accept any changes you might think it's useful. What I did:

1 - added IPR Praha logo, so you can use the same string to your next PRs
2 - added a description of the layer, so other people can understand what is low-vegetation
3 - added attribution and link to IPR Praha
4 - added "best": true given your feedback
5 - added "permission_osm": "explicit"
6 - "valid-georeference": true (I suspect it's true, and that everything in Praha is aligned to this layer, but please confirm)

Once you approve these (or any) of these suggestions, I'll merge the PR and you can work on the other layers based on this PR.

**Comment by cicku** [comment]
> To be honest, after doing the same thing in JOSM, I expected a quicker progress here. 

That is not how it works here. JOSM allows you to add any imagery without many checks, ELI rejects imageries with questionable licenses.

**Comment by Vlada-C** [comment]
Thanks, Matheus, for these improvements. I am fine with all of them, I hope I have managed to approve them the correct way.

> 3 - added attribution and link to IPR Praha

That's correct. The preferred source= value should be "CZ:IPRPraha:ortofoto"  according to [the talk-cz discussion](https://lists.openstreetmap.org/pipermail/talk-cz/2017-February/015901.html). I do not know if the attribution text is used for this purpose in iD for example? If so, "CZ:IPRPraha:ortofoto" could be perhaps better than "IPR Praha"

>  6 - "valid-georeference": true (I suspect it's true, and that everything in Praha is aligned to this layer, but please confirm)
 
I can confirm that the alignment is very accurate.

**Comment by matheusgomesms** [comment]
Thanks, all looking good for merging.

> That's correct. The preferred source= value should be "CZ:IPRPraha:ortofoto" according to [the talk-cz discussion](https://lists.openstreetmap.org/pipermail/talk-cz/2017-February/015901.html). I do not know if the attribution text is used for this purpose in iD for example? If so, "CZ:IPRPraha:ortofoto" could be perhaps better than "IPR Praha"

iD uses the name of the layer in the changeset in the `imagery_used=*` tag. I don't think `source` is populated with any info from ELI, only manually input from the mapper.

This value appears in the bottom left of the iD editor when editing (JOSM does the same I believe) and using this layer, so I think "IPR Praha" is a better value than "CZ:IPRPraha:ortofoto", but if you think it's not, please let me know and we'll switch before merging this PR.

**Comment by Vlada-C** [comment]
> This value appears in the bottom left of the iD editor when editing (JOSM does the same I believe) and using this layer, so I think "IPR Praha" is a better value than "CZ:IPRPraha:ortofoto", but if you think it's not, please let me know and we'll switch before merging this PR.

Thanks - let's stick with "IPR Praha" then.

**Pull request merged**